mlflow/shap.py

```diff
@@ -32,3 +41,142 @@ def _log_matplotlib_figure(fig, out_file, artifact_path=None):
     """
     with _log_artifact_contextmanager(out_file, artifact_path) as tmp_path:
         fig.savefig(tmp_path)
+
+
+@experimental
+def log_explanation(predict_function, features, artifact_path=None):
+    r"""
+    Given a ``predict_function`` capable of computing ML model output on the provided ``features``,
+    computes and logs explanations of an ML model's output. Explanations are logged as a directory
+    of artifacts containing the following items generated by `SHAP`_ (SHapley Additive
+    exPlanations).
+
+        - Base values
+        - SHAP values (computed using `shap.KernelExplainer`_)
+        - Summary bar plot (shows the average impact of each feature on model output)
+
+    :param predict_function:
+        A function to compute the output of a model (e.g. ``predict`` method of
```

Comment:
```suggestion
        A function to compute the output of a model (e.g. ``predict_proba`` method of scikit-learn classifiers
```

Tiny nit: could we update this example to specify that users need to use ``predict_proba`` for sklearn classifiers? Just because I think that's a bit subtler than the regressor case, where the only choice is to pass ``predict``.

Reply:
Makes sense. I'll udpate the example.